src/internal/internalAPIs.ts

```diff
@@ -284,6 +283,30 @@ function sendMessageResponse(
   }
 }
 
+/**
+ * Send a custom message object that can be sent to child window,
+ * instead of a response message to a child
+ */
+export function sendCustomMessageRequest(
+  targetWindow: Window | any,
```

Comment:
This API is used for sending an unsolicited custom message (id:null) from parent to child to it can be detected as meant for a handler and not a callback (and the child can register custom handlers)

Reply:
In that case, let's call this `sendCustomEvent`. Requests always go up to the parent, events go down to the child. Requests have an id and expect a response. Events don't have an id and are essentially fire and forget. Does that make sense? It's the same pattern that exists between Teams and its hosted iframes.